### PR TITLE
Upgrade conform version

### PR DESCRIPTION
**Pull request by edmundhung**
Upgraded conform to [v0.8](https://github.com/edmundhung/conform/releases/tag/v0.8.0) with schema adjusted for the new automatic type coercion feature :fire:

## Test Plan

I have test the following forms manually:
- Login (with 2fa)
- Forgot password / Reset password
- Onboarding
- Profile update
- Enable 2fa
- Update profile picture
- Create / Update Note

## Checklist

- [ ] Tests updated
- [ ] Docs updated

## Screenshots

n/a

### CODE REVIEW

File: app/routes/_auth+/onboarding.tsx [hunk 1]
```diff
@@ -38,24 +37,30 @@ const OnboardingFormSchema = z
...
-	})
+			.superRefine(({ confirmPassword, password }, ctx) => {
+				if (confirmPassword !== password) {
+					ctx.addIssue({
+						path: ['confirmPassword'],
+						code: 'custom',
+						message: 'The passwords did not match',
+					})
+				}
+			}),
```
Comment (edmundhung) [inline app/routes/_auth+/onboarding.tsx:62]: Zod does not run `refine` or `superRefine` until all fields are provided. This makes it run earlier as soon as both `password` and `confirmPassoword` is filled.

Ref: https://github.com/colinhacks/zod/issues/479

File: app/routes/_auth+/forgot-password/index.tsx [hunk 2]
```diff
@@ -90,7 +89,7 @@ export async function action({ request }: DataFunctionArgs) {
 	if (response.status === 'success') {
 		return redirect(redirectTo.toString())
 	} else {
-		submission.error[''] = response.error.message
+		submission.error[''] = [response.error.message]
```
Comment (edmundhung) [inline app/routes/_auth+/forgot-password/index.tsx:92]: The `report` helper is not available yet. But the error has to always be an array now.

Comment (kentcdodds) [reply app/routes/_auth+/forgot-password/index.tsx:92]: Sounds good 👍

File: app/routes/_auth+/onboarding.tsx (hunk 1 above)
Comment (kentcdodds) [reply app/routes/_auth+/onboarding.tsx:62]: Oh fantastic. Great idea. Thank you!

File: app/routes/resources+/note-editor.tsx [hunk 3]
```diff
@@ -14,16 +14,15 @@ import { redirectWithToast } from '~/utils/flash-session.server.ts'
 
 export const NoteEditorSchema = z.object({
 	id: z.string().optional(),
-	title: z.string().min(1),
-	content: z.string().min(1),
+	title: z.string(),
+	content: z.string(),
```
Comment (kentcdodds) [inline app/routes/resources+/note-editor.tsx:18]: I seem to remember talking about this, but I forgot. Does this mean that users will be able to create notes with no title or content? Or is `min(1)` assumed? If it's assumed, then I would need to do `z.string().optional()` if I wanted to allow an empty string? I think that makes sense in this context, but I just wanted to make sure my understanding is correct.

**Review by kentcdodds — APPROVED** [review]
Thank you! I'm so grateful to you for taking the time to contribute this upgrade!

**Comment by kentcdodds** [comment]
Will have to check in on why playwright's not working. I'll kick it off again to see if it was a fluke (😬).

File: app/routes/resources+/note-editor.tsx (hunk 3 above)
Comment (edmundhung) [reply app/routes/resources+/note-editor.tsx:18]: It's adhere to how zod works now which means all fields are required by default. 

>  I would need to do z.string().optional() if I wanted to allow an empty string
Exactly

**Comment by edmundhung** [comment]
Happy to see the zod schemas getting simplified with this! Love it. 😄 I pull the fix you made on main branch and the tests should be good now.

**Pull request merged**